packages/base-styles/_animations.scss

```diff
@@ -1,5 +1,5 @@
-@mixin edit-post__fade-in-animation($speed: 0.2s, $delay: 0s) {
-	animation: edit-post__fade-in-animation $speed ease-out $delay;
+@mixin edit-post__fade-in-animation($speed: 0.08s, $delay: 0s) {
```

Comment:
The way this mixin is defined has a few issues that we should solve:

- the mixin name suggests that the mixin is defined in the `@wordpress/edit-post` package, while it's actually defined in the `@wordpress/base-styles` package;
- the `edit-post__fade-in-animation` animation keyframes are defined (twice) in separate packages: [here](https://github.com/WordPress/gutenberg/blob/55f3a98159a754cee5162a05852881fa168134a7/packages/edit-post/src/style.scss#L13-L20) and [here](https://github.com/WordPress/gutenberg/blob/859d48b64dfc10c997f093ebea1a25d557c2f9c4/packages/edit-site/src/style.scss#L98-L108). That is a potential issue in itself, since:
  - these two animation definitions can get out of sync (ie. not have the same underlying keyframe specs)
  - if a third-party consumer of `@wordpress/base-styles` doesn't need to load the `@wordpress/edit-site` or `@wordpress/edit-post` packages, then the animation will not work (since it's not defined in `@wordpress/base-stlyles`)

The result is that, currently, the `Modal` component in `@wordpress/components` is consuming a SCSS mixin from `@wordpress/base-styles` (which is confusingly prefixed with the name of another package), but it needs for either the `@wordpress/edit-site` or the `@wordpress/edit-post` packages to be loaded too. This creates an implicit circular dependency that we should break.

---

Looking at [WPDirectory](https://wpdirectory.net/search/01J7DN5S7DZP3TC9ASEQKTCVNG), it looks like the `edit-post__fade-in-animation` mixin is used by third-party developers, and therefore we can't really rename it.

But we can still move the `@keyframes` definitions in the same file as the mixin, and delete the duplicate definitions in other packages — this will remove implicit circular dependencies, and prevent unexpected behaviors caused by the keyframes potentially having different definitions.

Reply:
Trying out the approach suggested above as part of https://github.com/WordPress/gutenberg/pull/65203#discussion_r1751968411

Reply:
Extracted to https://github.com/WordPress/gutenberg/pull/65377